aiida/schedulers/plugins/slurm.py

```diff
@@ -214,6 +214,19 @@ def _get_joblist_command(self, jobs=None, user=None):
                 if not isinstance(jobs, (tuple, list)):
                     raise TypeError("If provided, the 'jobs' variable must be a string or a list of strings")
                 joblist = jobs
+
+            # Trick: If we are asking for a single job, we append the same job once more.
+            # We do this because squeue returns a non-zero exit code if provided an invalid jobid
+            # (stderr: "slurm_load_jobs error: Invalid job id specified"), which can happen if time has passed since
+            # the job finished and squeue no longer has knowledge of it.
```

Comment:
Is it really when squeue has no longer knowledge or it, or even just if it has finished?

Reply:
You are right - i think it happens as soon as the job no longer is in the queue (will change!).